ncrypt/pgp.c

```diff
@@ -1663,22 +1641,21 @@ struct Body *pgp_class_traditional_encryptsign(struct Body *a, int flags, char *
   mutt_file_fclose(&pgpin);
 
   mutt_mktemp(pgpoutfile, sizeof(pgpoutfile));
-  mutt_mktemp(pgperrfile, sizeof(pgperrfile));
-  if ((pgpout = mutt_file_fopen(pgpoutfile, "w+")) == NULL ||
-      (pgperr = mutt_file_fopen(pgperrfile, "w+")) == NULL)
+  FILE *pgpout = mutt_file_fopen(pgpoutfile, "w+");
+  FILE *pgperr = mutt_file_mkstemp();
+  if (!pgpout || !pgperr)
   {
-    mutt_perror(pgpout ? pgperrfile : pgpoutfile);
+    mutt_perror(pgpout ? "mutt_file_mkstemp() failed!" : pgpoutfile);
     unlink(pgpinfile);
     if (pgpout)
     {
       mutt_file_fclose(&pgpout);
       unlink(pgpoutfile);
     }
+    mutt_file_fclose(&pgperr);
```

Comment:
I added this line.
If the `pgpoutfile` failed, we could leak the `pgperr` file handle.